packages/components/src/components/primitives/PinInput/PinInput.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC } from 'react';
+import {
+  Flex,
+  PinInput as ChakraPinInput,
+  PinInputField as ChakraPinInputField,
+  PinInputFieldProps,
+} from '@chakra-ui/react';
+import { PinInputProps, PinInputStaticMembers } from './types';
+
+const renderFieldsByNumOfDigits: (numOfDigits: number, pinInputFieldProps: PinInputFieldProps) => React.ReactNode[] = (
+  numOfDigits,
+  pinInputFieldProps
+) => {
+  const fields: React.ReactNode[] = [];
+  for (let digit = 0; digit < numOfDigits; digit++) {
+    fields.push(<ChakraPinInputField key={digit} {...pinInputFieldProps} />);
+  }
+  return fields;
+};
+
+export const PinInput: FC<PinInputProps> & PinInputStaticMembers = ({
+  children,
+  numOfDigits = 4,
+  placeholder = '0',
+  pinInputFieldProps,
+  ...rest
+}) => {
+  return (
+    <Flex justifyContent="center" columnGap="0.5rem">
+      <ChakraPinInput data-testid="cmpsr.pin-input" placeholder={placeholder} {...rest}>
+        {children || renderFieldsByNumOfDigits(numOfDigits, pinInputFieldProps)}
+      </ChakraPinInput>
+    </Flex>
+  );
+};
+
+const InputField: FC<PinInputFieldProps> = (props) => <ChakraPinInputField {...props} />;
```

Comment:
If this is the same as the `chakra` implementation we might want to remove this line and use the chakra component directly in `PinInput.Field = ChakraInputField` @TheEnmanuel23